examples/40_advanced/example_custom_ensemble.py

```diff
@@ -0,0 +1,781 @@
+            f"\tOOB Score: {self.get_validation_performance()}\n"
+            f"\tWeights: {self.weights_}\n"
+            f"\tIdentifiers: {identifiers_str}\n"
+        )
+
+    def get_models_with_weights(
+        self, models: Dict[Tuple[int, int, float], BasePipeline]
+    ) -> List[Tuple[float, BasePipeline]]:
+        output = []
+        for i, weight in enumerate(self.weights_):
+            if weight > 0.0:
+                identifier = self.identifiers_[i]
+                model = models[identifier]
+                output.append((weight, model))
+
+        output.sort(reverse=True, key=lambda t: t[0])
+
+        return output
+
+    def get_identifiers_with_weights(
+        self,
+    ) -> List[Tuple[Tuple[int, int, float], float]]:
+        return list(zip(self.identifiers_, self.weights_))
+
+    def get_selected_model_identifiers(self) -> List[Tuple[int, int, float]]:
+        output = []
+
+        for i, weight in enumerate(self.weights_):
+            identifier = self.identifiers_[i]
+            if weight > 0.0:
+                output.append(identifier)
+
+        return output
+
+    def get_validation_performance(self) -> float:
+        return self.validation_score_[0]
+
+    def get_pareto_front(self) -> Sequence[AbstractEnsemble]:
+        return_value = []
+        for i, ensemble in enumerate(self.ensembles_):
+            able = ABLE(
+                task_type=self.task_type,
+                metrics=self.metrics,
+                random_state=self.random_state,
+                backend=self.backend,
+                n_samples=self.n_samples,
+            )
+            able.weights_ = ensemble
+            able.identifiers_ = self.identifiers_
+            return_value.append(able)
+        return return_value
+
+
+import sklearn.datasets
+import sklearn.metrics
+import autosklearn.classification
+
+X, y = sklearn.datasets.fetch_openml(data_id=31, return_X_y=True, as_frame=True)
+# Change the target to align with scikit-learn's convention that
+# ``1`` is the minority class. In this example it is predicting
+# that a credit is "bad", i.e. that it will default.
+y = np.array([1 if val == "bad" else 0 for val in y])
+X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(
+    X, y, random_state=1
+)
+
+automl = autosklearn.classification.AutoSklearnClassifier(
+    time_left_for_this_task=60,
+    per_run_time_limit=10,
+    tmp_folder="/tmp/autosklearn_classification_example_tmp",
+    # ensemble_class=MultiObjectiveEnsembleWrapper,
+    # ensemble_kwargs={
+    #    'ensemble_class': ABLE,
+    #    'fidelity': 12,
+    #    'ensemble_kwargs': {
+    #        'n_samples': 5,
+    #    },
+    # },
+    # metric=[autosklearn.metrics.precision, autosklearn.metrics.recall],
+    # ensemble_class=MultiObjectiveABLE,
+    # ensemble_kwargs={"n_samples": 100, "n_weights": 1000},
+    metric=autosklearn.metrics.roc_auc,
+    # ensemble_class=ABLE,
+    # ensemble_kwargs={"n_samples": 200},
```

Comment:
The example doesn't use any of them? Perhaps it would be good to split these up into different examples as it's almost 800 lines of code for an example with multiple different custom ensembles. I didn't review the code for them since it's not core.

Reply:
I'm using this example mostly to play around myself and checked it in so others can have a look. Sorry, should have made that clear.